### ISSUE TITLE
unable to call function on mainnet fork due to outdated web3 dependency

### ISSUE THREAD

**Issue by npasquie** [NomicFoundation/hardhat#1380]
Found myself unable to call a read-only function on an already deployed smart-contract using hardhat-web3, mainnet forking and hardhat test. 

## How to reproduce : 

have hardhat-web3 installed

In the test folder :
find the sorareTokens ABI here https://etherscan.io/address/0x629A673A8242c2AC4B7B8C5D8735fbeac21A6205#code and put it into a `sorare.json` file.
create a script `test.js`
```
const sorareABI = require("./sorare.json")
const sorareTokensAddress = "0x629A673A8242c2AC4B7B8C5D8735fbeac21A6205"
const uniqueManuelCardId = '109885007871154280541989865417424574301301402155804365246179380903455247947907'

describe("using sorare contract", function (){
  it("should read ownerOf a NFT", async function (){
    const sorareTokens = await new web3.eth.Contract(sorareABI,sorareTokensAddress)
    await sorareTokens.methods.ownerOf(uniqueManuelCardId).call()
  })
})
```

launch a mainnet fork with `npx hardhat node --fork https://eth-mainnet.alchemyapi.io/v2/<yourKey>`
launch the test with `npx hardhat test --deploy-fixture`

the test will throw : `Error: Returned values aren't valid, did it run Out of Gas? You might also (...)`

## how to solve : 
Install the latest version of web3 through npm, then modify the test like this : 
```
const sorareABI = require("./sorare.json")
const Web3 = require("web3")
const sorareTokensAddress = "0x629A673A8242c2AC4B7B8C5D8735fbeac21A6205"
const uniqueManuelCardId = '109885007871154280541989865417424574301301402155804365246179380903455247947907'

describe("using sorare contract", function (){
  it("should read ownerOf a NFT", async function (){
    const myweb3 = await new Web3("http://localhost:8545")
    const sorareTokens = await new myweb3.eth.Contract(sorareABI,sorareTokensAddress)
    await sorareTokens.methods.ownerOf(uniqueManuelCardId).call()
  })
})
```

## proposed fix : 
This bug come form this issue : https://github.com/ChainSafe/web3.js/issues/1916#issuecomment-427398031
Of the web3.js package version 1.0.0-beta.36 used by hardhat-web3, it has been solved since, hardhat should update this dependency.

**Comment by alcuadrado** [comment]
Hey! Thanks for opening this issue. I'm a bit confused though, as Hardhat doesn't use web3.js 1.0.0-beta.36. You have to manually install web3, and most probably get the latest one.

**Comment by npasquie** [comment]
I have been using web3 by importing this package https://www.npmjs.com/package/@nomiclabs/hardhat-web3 (and not the web3js one) which includes web3 1.0.0-beta.36 as mentioned in this package.json : https://github.com/nomiclabs/hardhat/blob/master/packages/hardhat-web3/package.json 
so I hadn't imported web3js, I used the web3 object exposed by hardhat in tests

**Comment by fvictorio** [comment]
@alcuadrado maybe the peer dependency is being automatically installed?

@npasquie are you using npm or yarn? And which version are you using?

**Comment by npasquie** [comment]
I use yarn. I think that yarn installed the buggy version of web3js as it is mentioned as a peer dependency (didn't knew this existed lol), then I made a `yarn add -D web3` in my project, which is the version that was called when I made a `const myWeb3 = require("web3")` which fixed the problem. Thanks guys for the quick responses

**Comment by alcuadrado** [comment]
Which version of yarn do you use, @npasquie? It would be awesome if you can help us reproduce this.

**Comment by npasquie** [comment]
I use yarn v1.22.10. I am available on discord (nicos#4370) if needed

**Comment by fvictorio** [comment]
I can't reproduce this with the latest version of Hardhat and `@nomiclabs/hardhat-web3`.
